app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java

```diff
@@ -359,22 +367,37 @@ public boolean onMarkerClick(Marker marker) {
     }
 
     private void setupPOIListListener() {
-        mViewModel.getListOfPOI().observe(getViewLifecycleOwner(), new Observer<List<WalkingPoint>>() {
-            @Override
-            public void onChanged(List<WalkingPoint> walkingPoints) {
-                for (WalkingPoint point : walkingPoints) {
-                    LatLng latLng = new LatLng(point.getCoordinate().getLongitude(), point.getCoordinate().getLatitude());
-                    MarkerOptions markerOptions = new MarkerOptions()
-                            .position(latLng)
-                            //TODO: Change the icon to be the same as the POI icon
-                            .icon(BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_GREEN))
-                            .visible(false);
-                    mMap.addMarker(markerOptions);
+        mViewModel.getListOfPOI().observe(getViewLifecycleOwner(), priorityQueue -> {
+            int position = 1;
```

Comment:
is there a reason why this is a do while and not a for loop?

Reply:
We can only poll a priority queue (heap). It's not easily iterable.

Reply:
Ok, can you add a comment to explain this decision? will help understand this code down the line

Reply:
Actually, we realized that this can be a for loop. It's just a question of choice.